src/frontend-samples/display-styles-sample/DisplayStylesUI.tsx

```diff
@@ -0,0 +1,106 @@
+import { IModelApp, IModelConnection, ScreenViewport, Viewport } from "@bentley/imodeljs-frontend";
+import "common/samples-common.scss";
+import { ControlPane } from "Components/ControlPane/ControlPane";
+import { ReloadableViewport } from "Components/Viewport/ReloadableViewport";
+import * as React from "react";
+import DisplayStylesApp from "./DisplayStylesApp";
+import { DisplayStyle } from "./Styles";
+import { Toggle } from "@bentley/ui-core";
+
+interface DisplayStylesUIState {
+  activePresetIndex: number;
+  merge: boolean;
+  viewport?: Viewport;
+}
+
+interface DisplayStylesUIProps {
+  iModelName: string;
+  iModelSelector: React.ReactNode;
+  displayStyles: DisplayStyle[];
+}
+
+const CUSTOM_STYLE_INDEX = 0;
+
+export default class DisplayStylesUI extends React.Component<DisplayStylesUIProps, DisplayStylesUIState> {
+
+  public state: DisplayStylesUIState = { activePresetIndex: 2, merge: false };
+
+  // Called by the control and will update the active display style.
+  private readonly _onChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const index = Number.parseInt(event.target.value, 10);
+    this.setState({ activePresetIndex: index });
+  }
+
+  // Called by the control and updates wether to also apply the Custom display style.
+  private readonly _onToggle = (isOn: boolean) => {
+    this.setState({ merge: isOn });
+  }
+
+  // Will be triggered once when the iModel is loaded.
+  private readonly _onIModelReady = (_iModel: IModelConnection) => {
+    const vp = IModelApp.viewManager.selectedView;
+    if (undefined === vp)
+      IModelApp.viewManager.onViewOpen.addOnce((viewport: ScreenViewport) => {
+        this.setState({ viewport });
+      });
+    else
+      this.setState({ viewport: vp });
+  }
+
+  /** A render method called when the state or props are changed. */
+  public componentDidUpdate(_prevProp: DisplayStylesUIProps, prevState: DisplayStylesUIState) {
+    let updateDisplayStyle = false;
+    if (this.state.viewport?.viewportId !== prevState.viewport?.viewportId)
+      updateDisplayStyle = true;
+    if (this.state.activePresetIndex !== prevState.activePresetIndex)
+      updateDisplayStyle = true;
+    if (this.state.merge !== prevState.merge)
+      updateDisplayStyle = true;
+
+    if (updateDisplayStyle && undefined !== this.state.viewport) {
+      const style = this.props.displayStyles[this.state.activePresetIndex];
+      DisplayStylesApp.applyDisplayStyle(this.state.viewport, style);
+      if (this.state.merge && CUSTOM_STYLE_INDEX !== this.state.activePresetIndex) {
+        const custom = this.props.displayStyles[CUSTOM_STYLE_INDEX];
+        DisplayStylesApp.applyDisplayStyle(this.state.viewport, custom);
+      }
+    }
+  }
+
+  private getControls(): React.ReactNode {
+    const toggleTooltip = "Toggling on will apply the \"Custom\" style in \"Styles.ts\" after apply the selected style.";
```

Comment:
Grammar:
Toggling on will apply the \"Custom\" style in \"Styles.ts\" after the selected style is applied.